compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -852,9 +852,22 @@ pub fn build_compile_unit_di_node<'ll, 'tcx>(
     // FIXME(#41252) Remove "clang LLVM" if we can get GDB and LLVM to play nice.
     let producer = format!("clang LLVM ({rustc_producer})");
 
-    use rustc_session::RemapFileNameExt;
     let name_in_debuginfo = name_in_debuginfo.to_string_lossy();
-    let work_dir = tcx.sess.opts.working_dir.for_codegen(tcx.sess).to_string_lossy();
+    debug!(?name_in_debuginfo, "build_compile_unit_di_node");
+
+    // Path of working directory in the root DI node seems to be embedded in
+    // executables. Hence, we trim them when the scope of `unsplit-debuginfo`
+    // is present, as if it is kinda a "unsplit" debuginfo.
```

Comment:
I've been able to verify that this is indeed the case for DWARF dwps (i.e. the DW_AT_comp_dir is only contained in the binary, not in the dwp). However, I'm not sure that this behavior is guaranteed, so I think we should test that the `dwp` and `dwo` files don't contain the compiler's working directory at all.